internal/provider/resources/resource_team.go

```diff
@@ -0,0 +1,602 @@
+			"Client Error",
+			fmt.Sprintf("Unable to mutate Team roles and list deployments, got error: %s", err),
+		)
+		return diags
+	}
+	_, diagnostic := clients.NormalizeAPIError(ctx, listDeployments.HTTPResponse, listDeployments.Body)
+	if diagnostic != nil {
+		diags.Append(diagnostic)
+		return diags
+	}
+
+	// get list of workspace ids from deployments
+	deploymentWorkspaceIds := lo.Map(listDeployments.JSON200.Deployments, func(deployment platform.Deployment, _ int) string {
+		return deployment.WorkspaceId
+	})
+
+	// get list of workspaceIds
+	workspaceIds := lo.Map(workspaceRoles, func(role iam.WorkspaceRole, _ int) string {
+		return role.WorkspaceId
+	})
+
+	// check if deploymentWorkspaceIds are in workspaceIds
+	missingWorkspaceIds, _ := lo.Difference(lo.Uniq(deploymentWorkspaceIds), lo.Uniq(workspaceIds))
+	if len(missingWorkspaceIds) > 0 {
+		tflog.Error(ctx, "failed to mutate Team roles", map[string]interface{}{
+			"error": fmt.Sprintf(
+				"Not every deployment role has a corresponding workspace role. Missing workspace roles for deploymentWorkspaceIds: %v", missingWorkspaceIds),
+		})
+		diags.AddError(
+			"Bad Request Error",
+			fmt.Sprintf("Unable to mutate Team roles, not every deployment role has a corresponding workspace role. Add workspace roles for the missing workspace ids: %v", missingWorkspaceIds),
+		)
+		return diags
+	}
+
+	// create request
+	updateTeamRolesRequest := iam.UpdateTeamRolesJSONRequestBody{
+		DeploymentRoles:  &deploymentRoles,
+		OrganizationRole: iam.UpdateTeamRolesRequestOrganizationRole(data.OrganizationRole.ValueString()),
+		WorkspaceRoles:   &workspaceRoles,
+	}
+	teamRoles, err := r.IamClient.UpdateTeamRolesWithResponse(
+		ctx,
+		r.OrganizationId,
+		teamId,
+		updateTeamRolesRequest,
+	)
+	if err != nil {
+		tflog.Error(ctx, "failed to mutate Team roles", map[string]interface{}{"error": err})
+		diags.AddError(
+			"Client Error",
+			fmt.Sprintf("Unable to mutate Team roles, got error: %s", err),
+		)
+		return diags
+	}
+	_, diagnostic = clients.NormalizeAPIError(ctx, teamRoles.HTTPResponse, teamRoles.Body)
+	if diagnostic != nil {
+		diags.Append(diagnostic)
+		return diags
+	}
+
+	return nil
+}
+
+// Create a team - uses createTeam and updateTeamRoles
+func (r *TeamResource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	var data models.TeamResource
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var diags diag.Diagnostics
+
+	memberIds, diags := utils.TypesSetToStringSlice(ctx, data.MemberIds)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	// Create the team request
+	createTeamRequest := iam.CreateTeamRequest{
+		Name:        data.Name.ValueString(),
+		Description: data.Description.ValueStringPointer(),
+		MemberIds:   &memberIds,
+	}
+
+	if !data.OrganizationRole.IsNull() {
```

Comment:
shuold never be null so prob dont need this and can add it to the initialization